### PR TITLE
Fix sweg and gainz routes. Added some assertions to get rid of some Pydantic errors/warnings.

### PR DESCRIPTION
**Pull request by ThorntonMatthewD**

### CODE REVIEW

**Review by Ferrariic — APPROVED** [review]

**Review by Ferrariic — APPROVED** [review]

File: api/routers/legacy.py [hunk 1]
```diff
@@ -124,7 +124,8 @@ async def sql_insert_player(player_name):
 
     await execute_sql(sql_insert, param=param)
     player = await sql_get_player(player_name)
-    return player
+
+    return player[0] if player is not None else None
```
Comment (extreme4all) [inline api/routers/legacy.py:128]: i think this is more readable?
`return None if player is None else player[0]`
or 
`return player[0] if not player is None else None`

**Comment by extreme4all** [comment]
i would advise against commiting any new features to legacy.
this feature also affects many other routes

File: api/routers/legacy.py [hunk 2]
```diff
@@ -1043,14 +1058,17 @@ async def get_latest_xp_gains(player_info:PlayerName, token:str):
...
+    players = await sql_get_player(player_name)
+
+    if players is None:
         raise HTTPException(404, detail="Player not found.")
 
-    player_id = player.get('id')
+    player_ids = [p.get('id') for p in players]
 
-    last_xp_gains = await sql_get_latest_xp_gain(player_id)
+    last_xp_gains = await sql_get_latest_xp_gain(player_ids)
```
Comment (extreme4all) [inline api/routers/legacy.py:1071]: the logic below does not work with multiple player_ids

File: api/routers/legacy.py [hunk 3]
```diff
@@ -1043,14 +1058,17 @@ async def get_latest_xp_gains(player_info:PlayerName, token:str):
     player = player_info.dict()
     player_name = player.get('player_name')
 
-    player = await sql_get_player(player_name)
+    if not player_name:
+        raise HTTPException(status_code=400, detail=f"player_name is a required parameter.")
```
Comment (extreme4all) [inline api/routers/legacy.py:1062]: add a logger.debug()

File: api/routers/legacy.py [hunk 4]
```diff
@@ -1176,6 +1196,10 @@ async def get_region(token:str, region: RegionName):
     region_info = region.dict()
     region_name = region_info.get('region_name')
 
+    if region_name is None:
```
Comment (extreme4all) [inline api/routers/legacy.py:1199]: this should already been validated by fast api

**Review by extreme4all — CHANGES_REQUESTED** [review]
the logic of the routes where you use a list of players does not work  with a list of players

**Pull request closed without merging**